front/components/vaults/VaultCreateWebsiteModal.tsx

```diff
@@ -1,32 +1,602 @@
+    if (nameExists) {
+      nameError = "A Folder with the same name already exists";
+    } else if (!dataSourceName.length) {
+      nameError = "Please provide a name.";
+    } else if (dataSourceNameRes.isErr()) {
+      nameError = dataSourceNameRes.error;
+    }
+
+    setDataSourceUrlError(urlError);
+    setDataSourceNameError(nameError);
+    return !urlError && !nameError;
+  }, [dataSourceName, dataSources, dataSourceUrl, dataSourceOrView?.id]);
+
+  useEffect(() => {
+    if (isSubmitted) {
+      validateForm();
+    }
+  }, [dataSourceName, dataSourceUrl, isSubmitted, validateForm]);
+
+  const handleCreate = async () => {
+    setIsSubmitted(true);
+
+    if (!validateForm()) {
+      return;
+    }
+
+    setIsSaving(true);
+    if (webCrawlerConfiguration === null) {
+      const sanitizedDataSourceUrl = dataSourceUrl.trim();
+      const res = await fetch(
+        `/api/w/${owner.sId}/vaults/${vault.sId}/data_sources/managed`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            provider: "webcrawler",
+            connectionId: "none",
+            name: dataSourceName,
+            configuration: {
+              url: sanitizedDataSourceUrl,
+              maxPageToCrawl: maxPages || WEBCRAWLER_MAX_PAGES,
+              depth: maxDepth,
+              crawlMode: crawlMode,
+              crawlFrequency: selectedCrawlFrequency,
+              headers: headers.reduce(
+                (acc, { key, value }) => {
+                  if (key && value) {
+                    acc[key] = value;
+                  }
+                  return acc;
+                },
+                {} as Record<string, string>
+              ),
+            } satisfies WebCrawlerConfigurationType,
+          } satisfies t.TypeOf<typeof PostManagedDataSourceRequestBodySchema>),
+        }
+      );
+      if (res.ok) {
+        setOpen(false);
+        sendNotification({
+          title: "Website created",
+          type: "success",
+          description: "The website has been successfully created.",
+        });
+        await mutateVaultDataSourceViews();
+        const response: PostVaultDataSourceResponseBody = await res.json();
+        const { dataSourceView } = response;
+        await router.push(
+          `/w/${owner.sId}/data-sources/vaults/${vault.sId}/categories/${WEBSITE_CAT}/data_source_views/${dataSourceView.sId}`
+        );
+      } else {
+        const err = (await res.json()) as { error: APIError };
+        setIsSaving(false);
+        sendNotification({
+          title: "Error creating website",
+          type: "error",
+          description: err.error.message,
+        });
+      }
+    } else if (dataSourceOrView) {
+      // TODO(GROUPS_INFRA): this should be refactored to use a patch route under /vaults
+      const res = await fetch(
+        `/api/w/${owner.sId}/data_sources/${encodeURIComponent(
```

Comment:
We have this endpoint now, right?

Reply:
nope not yet, we have only the post and delete in vault